src/components/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+import styles from './CheckBox.module.scss';
+
+export type CheckBoxProps = Omit<
+  React.InputHTMLAttributes<HTMLInputElement>,
+  'onChange'
+> & {
+  /** Вызывается при клике на чекбокс */
+  onChange: (value: boolean) => void;
+};
+
+export const CheckBox: React.FC<CheckBoxProps> = ({
+  checked,
+  onChange,
+  disabled,
+  ...props
+}) => {
+  return (
+    <input
+      className={styles.checkbox}
+      type="checkbox"
+      disabled={disabled}
+      checked={checked}
+      onClick={disabled ? () => {} : () => onChange(!checked as boolean)}
```

Comment:
вероятно, лучше вместо пропа onClick использовать проп onChange, куда бы приходило событие изменения состояния чекбокса и откуда бы доставалось значение checked, вместо полагания на внешнее состояние checked:
```suggestion
      onchange={disabled ? undefined : (event) => onChange(event.target.checked)}
```